core/runtime.rs

```diff
@@ -480,70 +575,58 @@ impl JsRuntime {
       state.known_realms.push(global_realm);
     }
     scope.set_data(
-      Self::STATE_DATA_OFFSET,
+      STATE_DATA_OFFSET,
       Rc::into_raw(state_rc.clone()) as *mut c_void,
     );
     let module_map_rc = Rc::new(RefCell::new(ModuleMap::new(loader, op_state)));
-    if let Some(snapshotted_data) = maybe_snapshotted_data {
+    if let Some(snapshotted_data) = snapshotted_data {
       let mut module_map = module_map_rc.borrow_mut();
       module_map.update_with_snapshotted_data(scope, snapshotted_data);
     }
     scope.set_data(
-      Self::MODULE_MAP_DATA_OFFSET,
+      MODULE_MAP_DATA_OFFSET,
       Rc::into_raw(module_map_rc.clone()) as *mut c_void,
     );
 
     drop(context_scope);
 
-    let mut js_runtime = Self {
-      v8_isolate: Some(isolate),
-      snapshot_options,
-      snapshot_module_load_cb: options.snapshot_module_load_cb.map(Rc::new),
+    let mut js_runtime = JsRuntimeImpl {
+      inner: InnerIsolateState {
+        snapshotting,
+        state: ManuallyDropRc(ManuallyDrop::new(state_rc)),
+        v8_isolate: ManuallyDrop::new(isolate),
+      },
+      bindings_mode,
       allocations: IsolateAllocations::default(),
-      event_loop_middlewares: Vec::with_capacity(options.extensions.len()),
-      extensions: Rc::new(RefCell::new(options.extensions)),
-      state: state_rc,
+      event_loop_middlewares,
+      extensions: options.extensions,
       module_map: Some(module_map_rc),
       is_main: options.is_main,
     };
 
-    // Init resources and ops before extensions to make sure they are
-    // available during the initialization process.
-    js_runtime.init_extension_ops().unwrap();
     let realm = js_runtime.global_realm();
-    js_runtime.init_extension_js(&realm).unwrap();
-
+    // TODO(mmastrac): We should thread errors back out of the runtime
```

Comment:
I think it's fine to unwrap here - if something goes wrong it suggests coding error on the embedder side

Reply:
I think we should fix this at some point because the errors are pretty bad -- once in a while VSCode creates an invalid import in one of our core libs and the build fails with a very strange panic (ie: files not consumed by the ExtModuleLoader, plus a bit backtrace).

Reply:
Alright, I guess that makes sense after all.